trivial: Mark overrides as such.

domob1812 opened the pull request:
This trivial change adds the `override` keyword to some methods that override virtual base class / interface methods.  This ensures that any future changes to the interface's method signatures which are not correctly mirrored in the subclasses will break at compile time with a clear error message, rather than at runtime.

GreatSock commented:
ACK

sipa commented:
You can also include `ConsolePrinter` and `PlotlyPrinter` in src/bench/bench.h.

domob1812 commented:
Indeed, added the printers in a second commit.  (I actually ran into the test failure while maintaining Namecoin.  But including more of these is of course even better.)

fanquake commented:
@domob1812 Can you please squash your commits.

domob1812 committed:
trivial: Mark overrides as such.

This trivial change adds the "override" keyword to some methods of
subclasses meant to override interface methods.  This ensures that any
future change to the interface' method signatures which are not correctly
mirrored in the subclass will break at compile time with a clear error message,
rather than fail at runtime (which is harder to debug).

domob1812 commented:
Squashed the commits.

Empact commented:
utACK 60ebc7d

fanquake commented:
utACK 60ebc7d
